common/src/query.rs

```diff
@@ -0,0 +1,137 @@
+// Types related to QueryList and Watch API
+use crate::pb;
+
+use std::str::FromStr;
+use std::string::ToString;
+
+use anyhow::{anyhow, bail};
+use serde::Serialize;
+use strum::{Display, EnumString};
+
+#[derive(Debug, Clone, Copy, Serialize, EnumString, Display)]
+pub enum VMStatus {
+    Running,
+    PoweredOff,
+    Paused,
+}
+
+impl Default for VMStatus {
+    fn default() -> Self {
+        VMStatus::Running
+    }
+}
+
+#[derive(Debug, Clone, Copy, Serialize, EnumString, Display)]
+pub enum TrustLevel {
+    Secure,
+    Warning,
+    NotSecure,
+}
+
+impl Default for TrustLevel {
+    fn default() -> Self {
+        TrustLevel::Warning
+    }
+}
+
+#[derive(Debug, Clone, Serialize)]
+pub struct QueryResult {
+    pub name: String,        //VM name
+    pub description: String, //App name, some details
+    pub status: VMStatus,
+    pub trust_level: TrustLevel,
+}
+
+impl QueryResult {
+    pub fn parse_list(items: Vec<pb::QueryListItem>) -> anyhow::Result<Vec<QueryResult>> {
+        let result = items
+            .into_iter()
+            .map(Self::try_from)
+            .collect::<anyhow::Result<Vec<QueryResult>>>()?;
+        Ok(result)
+    }
+}
+
+impl TryFrom<pb::QueryListItem> for QueryResult {
+    type Error = anyhow::Error;
+    fn try_from(item: pb::QueryListItem) -> Result<QueryResult, Self::Error> {
+        Ok(QueryResult {
+            name: item.name,
+            description: item.description,
+            status: VMStatus::from_str(item.vm_status.as_str())?,
+            trust_level: TrustLevel::from_str(item.trust_level.as_str())?,
+        })
+    }
+}
+
+impl Into<pb::QueryListItem> for QueryResult {
```

Comment:
`clippy` suggests implementing `From` over `Into` (because then you get `Into` for free)

Reply:
We can't  implement From<QueryResult> for pb:* because all pb::* stuff is protobuf generated, it why I choose Into here